### PR TITLE
Allow custom service name in CloudFoundry VCAP_SERVICES env var

### LINKED ISSUE #246: Allow VCAP service name to be specified
Allow service names other than the hard coded default `cloudantNoSQLDB`.

### PR DESCRIPTION
**Pull request by smithsz**
Thanks for your hard work, please ensure all items are complete before opening.

- [x] Tick to sign-off your agreement to the [Developer Certificate of Origin (DCO) 1.1](https://github.com/cloudant/nodejs-cloudant/blob/master/DCO1.1.txt)
- [x] You have added tests for any code changes
- [x] You have updated the [CHANGES.md](https://github.com/cloudant/nodejs-cloudant/blob/master/CHANGES.md)
- [x] You have completed the PR template below:

## What
Allow custom service name in CloudFoundry `VCAP_SERVICES` env var.

## How
Override default using `config.vcapServiceName`.

## Testing
Includes additional unit tests.

## Issues
Fixes #246.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -149,10 +149,12 @@ Running on Bluemix? You can initialize Cloudant directly from the `VCAP_SERVICES
 
 ~~~ js
 var Cloudant = require('cloudant');
-var cloudant = Cloudant({instanceName: 'foo', vcapServices: JSON.parse(process.env.VCAP_SERVICES)});
+var cloudant = Cloudant({ vcapInstanceName: 'foo', vcapServices: JSON.parse(process.env.VCAP_SERVICES) });
```
Comment (tomblench) [inline README.md]: I see that `instanceName` can be still be used as an alias for `vcapInstanceName`, this should probably be mentioned somewhere. Initially when I saw this in the README I assumed it was a breaking change.

Comment (smithsz) [reply README.md]: Added note about alias `instanceName`.

File: test/legacy/reconfigure.js [hunk 2]
```diff
@@ -141,6 +141,31 @@ describe('Reconfigure', function() {
     done();
   });
 
+  it('gets URL by instance name alias from vcap containing single service', function(done) {
+    var config = { vcapInstanceName: 'serviceA', // alias of 'instanceName'
+      vcapServices: { cloudantNoSQLDB: [
+        { name: 'serviceA', credentials: { url: 'http://mykey:mypassword@mydomain.cloudant.com' } }
```
Comment (emlaver) [inline test/legacy/reconfigure.js]: Should we have the same `vcapServices` object design as the one in java-cloudant with the URI, username, and password?
https://github.com/cloudant/java-cloudant/blob/master/cloudant-client/src/test/java/com/cloudant/tests/CloudFoundryServiceTest.java#L66

Comment (smithsz) [reply test/legacy/reconfigure.js]: Yeah, it'd be nice but there are quite a few of these already. I reckon we should do this when we need to change the vcap stuff for IAM.

Comment (emlaver) [reply test/legacy/reconfigure.js]: Ok sounds good.

**Pull request merged**